fix(Carousel): resolve plugins with page transitions

mikenewbon opened the pull request:
- Replace computedAsync with manual plugin loading using ref + watch
- Add retry logic with exponential backoff for dynamic import failures
- Fix fade transitions breaking after page navigation with pageTransition enabled
- Maintain all existing plugin functionality while improving reliability

Fixes issue where carousel fade property stopped working after page navigation when page transitions were configured in Nuxt applications.

### 🔗 Linked issue

Resolves #4379 

### ❓ Type of change

- [ ] 📖 Documentation (updates to the documentation or readme)
- [x] 🐞 Bug fix (a non-breaking change that fixes an issue)
- [ ] 👌 Enhancement (improving an existing functionality)
- [ ] ✨ New feature (a non-breaking change that adds functionality)
- [ ] 🧹 Chore (updates to the build process or auxiliary tools and libraries)
- [ ] ⚠️ Breaking change (fix or feature that would cause existing functionality to change)

### 📚 Description

The Nuxt UI Carousel component's `fade` property would stop working after page navigation when `pageTransition` was enabled in Nuxt applications. This bug manifested as:

- Fade transitions worked correctly on initial page load
- After navigating between pages, fade functionality broke completely
- Carousel would revert to slide animations instead of fade transitions
- No console errors were displayed, making it difficult to debug

This issue only occurred when both conditions were met:
1. Carousel component had `fade` prop enabled
2. Page transitions were configured using `definePageMeta({ pageTransition: ... })`

#### Root Cause Analysis

The bug was caused by **VueUse's `computedAsync` failing during page transitions** in Nuxt:

1. **Dynamic Import Timing Issues**: The carousel used `computedAsync` to dynamically import Embla Carousel plugins (including `embla-carousel-fade`)
2. **Page Transition Interference**: When page transitions occurred, the Vue component context was torn down and recreated
3. **Silent Failures**: During this transition, `computedAsync`'s dynamic imports would fail silently
4. **Plugin Loss**: The fade plugin failed to load, causing `plugins.value` to become an empty array
5. **Carousel Reinitializes**: The carousel would reinitialize without the fade plugin, reverting to slide behavior

#### Technical Solution

Replaced the unreliable `computedAsync` approach with a robust manual implementation:

##### Before (Problematic):
```typescript
const plugins = computedAsync<EmblaPluginType[]>(async () => {
  // Dynamic imports that failed during page transitions
  if (props.fade) {
    const FadePlugin = await import('embla-carousel-fade').then(r => r.default)
    plugins.push(FadePlugin(/* options */))
  }
  return plugins
})
```

##### After (Fixed):
```typescript
const plugins = ref<EmblaPluginType[]>([])

const loadPlugins = async (retryCount = 0) => {
  try {
    // Same plugin loading logic with retry mechanism
    if (props.fade) {
      const FadePlugin = await import('embla-carousel-fade').then(r => r.default)
      pluginList.push(FadePlugin(/* options */))
    }
    plugins.value = pluginList
  } catch (error) {
    // Retry with exponential backoff
    if (retryCount < 3) {
      setTimeout(() => loadPlugins(retryCount + 1), Math.pow(2, retryCount) * 100)
    }
  }
}

// Load on mount and when props change
onMounted(() => loadPlugins())
watch([/* plugin props */], () => loadPlugins())
```

### 📝 Checklist

- [x] I have linked an issue or discussion.
- [x] I have updated the documentation accordingly.

mikenewbon committed:
fix(carousel): resolve fade plugin failure during page transitions

- Replace computedAsync with manual plugin loading using ref + watch
- Add retry logic with exponential backoff for dynamic import failures
- Fix fade transitions breaking after page navigation with pageTransition enabled
- Maintain all existing plugin functionality while improving reliability

Fixes issue where carousel fade property stopped working after page navigation
when page transitions were configured in Nuxt applications.

mikenewbon commented:
Just want to say, what a cool project to contribute to! It's my first time here and I'm a little worried my solution is a bandaid and I should have followed the breadcrumbs further back - but I really appreciate this pipeline that lets me install the built package on my project. Hope I can contribute further in the future.

benjamincanac reviewed (commented):
I understand the issue with `computedAsync` which would be enough to solve with a function instead but what's the point of retrying? 🤔

mikenewbon committed:
refactor(carousel): simplify plugin loading solution

Remove unnecessary retry logic and error handling complexity.
Replace computedAsync with simple async function approach as suggested by Nuxt team.

Maintains the core fix for fade plugin breaking after page transitions.

mikenewbon committed:
refactor(carousel): further simplify plugin loading

Remove unnecessary plugin prop watcher and consolidate onMounted calls.
The existing options/plugins watcher handles reinitialization correctly.

mikenewbon commented:
@benjamincanac I shouldnt make commits at 4am, I've cleaned up a little, hope thats better!

mikenewbon committed:
refactor(carousel): further further simplify plugin loading

Remove intermediate pluginList variable and push directly to plugins.value.

mikenewbon committed:
fix(carousel): revert to working pluginList implementation

The direct plugins.value.push() approach broke the fade functionality.
Restore the pluginList approach that properly fixes the page transition bug.

benjamincanac commented on src/runtime/components/Carousel.vue:
```diff
@@ -265,6 +267,8 @@ function isCarouselItem(item: CarouselItem): item is _CarouselItem {
 }
 
 onMounted(() => {
+  loadPlugins()
```
I would keep a watch immediate on the props to load the plugins instead, it's no longer reactive.

mikenewbon replied:
My latest commit should resolve.

mikenewbon committed:
Add back reactivity by watching props.

benjamincanac committed:
Merge branch 'v3' into fix/carousel-fade-page-transitions

benjamincanac committed:
up

benjamincanac reviewed (approved):
I've removed the `loadPlugins` call from `onMounted` since it was called twice with the immediate watch.
